### PR TITLE
[nrf fromtree] net: wifi_mgmt: Display TWT setup response status

### PR DESCRIPTION
**Pull request by ajayparida**
Display the TWT setup response status.

Signed-off-by: Ajay Parida <ajay.parida@nordicsemi.no>
(cherry picked from commit c69731a281492ec592fdf5485deb8b3edd5f5997)

### CODE REVIEW

File: include/zephyr/net/wifi.h [hunk 1]
```diff
@@ -318,4 +318,15 @@ static const char * const wifi_twt_setup_cmd2str[] = {
...
 
+enum wifi_twt_setup_resp_status {
+	/* TWT Setup response status */
+	WIFI_TWT_RESP_NOT_RECEIVED = 0,
+	WIFI_TWT_RESP_PROCESSING,
+};
+
+static const char * const wifi_twt_setup_resp2str[] = {
+	/* TWT Setup response status */
+	[WIFI_TWT_RESP_NOT_RECEIVED] = "TWT response not received",
```
Comment (krish2718) [inline include/zephyr/net/wifi.h:329]: WIFI_TWT_RESP_PROCESSING?

File: subsys/net/l2/wifi/wifi_shell.c [hunk 2]
```diff
@@ -153,16 +153,21 @@ static void handle_wifi_twt_event(struct net_mgmt_event_callback *cb)
...
-
-	/* If accepted, then no need to print TWT params */
-	if (resp->setup_cmd != WIFI_TWT_SETUP_CMD_ACCEPT) {
-		print(context.sh, SHELL_NORMAL,
-		      "TWT parameters: trigger: %s wake_interval_ms: %d, interval_ms: %d\n",
-		      resp->setup.trigger ? "trigger" : "no_trigger",
-		      resp->setup.twt_wake_interval_ms,
-		      resp->setup.twt_interval_ms);
+	if (resp->resp_status == WIFI_TWT_RESP_NOT_RECEIVED) {
+		print(context.sh, SHELL_NORMAL, "TWT response status: %s\n",
```
Comment (krish2718) [inline subsys/net/l2/wifi/wifi_shell.c:157]: This doesn't make sense, response status but response not received? we need to fix terminology b/w RPU status and TWT response status. Something like:

if (rpu_stats == WIFI_TWT_RESP_NOT_RECEIVED)
   print ("TWT response timed out")

**Review by krish2718 — CHANGES_REQUESTED** [review]

File: include/zephyr/net/wifi.h (hunk 1 above)
Comment (ajayparida) [reply include/zephyr/net/wifi.h:329]: It's not required as already print message for this WIFI_TWT_RESP_PROCESSING is there.

**Review by krish2718 — APPROVED** [review]

File: subsys/net/l2/wifi/wifi_shell.c [hunk 3]
```diff
@@ -153,16 +153,20 @@ static void handle_wifi_twt_event(struct net_mgmt_event_callback *cb)
...
-		      resp->setup.twt_wake_interval_ms,
-		      resp->setup.twt_interval_ms);
+	if (resp->resp_status == WIFI_TWT_RESP_RECEIVED) {
+		print(context.sh, SHELL_NORMAL, "TWT response: %s for dialog: %d and flow: %d\n",
+		      wifi_twt_setup_cmd2str[resp->setup_cmd], resp->dialog_token, resp->flow_id);
+
+		/* If accepted, then no need to print TWT params */
+		if (resp->setup_cmd != WIFI_TWT_SETUP_CMD_ACCEPT) {
+			print(context.sh, SHELL_NORMAL,
+			      "TWT parameters: trigger: %s wake_interval_ms: %d, interval_ms: %d\n",
```
Comment (sachinthegreen) [inline subsys/net/l2/wifi/wifi_shell.c:163]:
```suggestion
			      "TWT parameters: trigger: %s wake_interval: %d ms, interval: %d ms\n",
```

**Review by sachinthegreen — CHANGES_REQUESTED** [review]

**Review by jfischer-no — CHANGES_REQUESTED** [review]
These commits are part of current upmerge, can this PR be closed?
yeah, I will close it

**Pull request closed without merging**